lib/Dredd.js

```diff
@@ -255,18 +258,26 @@ Is the provided path correct?
       const fileData = this.configuration.data[filename];
       if (!fileData.annotations) { fileData.annotations = []; }
 
-      logger.debug('Compiling HTTP transactions from API description file:', filename);
-      dreddTransactions.compile(fileData.raw, filename, (compilationError, compilationResult) => {
-        if (compilationError) { return next(compilationError); }
-
-        fileData.mediaType = compilationResult.mediaType;
-        fileData.annotations = fileData.annotations.concat(compilationResult.annotations);
-        this.transactions = this.transactions.concat(compilationResult.transactions);
+      this.logger.debug(`Parsing API description: ${filename}`);
+      parse(fileData.raw, (parseErr, parseResult) => {
```

Comment:
This section is a bit messy, but works for now. I plan to continue refactoring this code further to make it more readable and testable, but didn't want to snowball everything into a single PR.